makefile

```diff
@@ -0,0 +1,63 @@
+GEN_DIR := ./.gen
```

Comment:
I might be wrong but Makefiles seem to be very rare in the JS/TS world (haven't seen one so far). This is more something that is common in the C, Go, C++ space, isn't it? 
Shouldn't we express these steps with "common" JS tools (package.json or maybe a scripts folder)?